consensus/poet/core/tests/test_consensus/test_poet_block_publisher.py

```diff
@@ -0,0 +1,765 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import shutil
+import tempfile
+
+from importlib import reload
+
+from unittest import TestCase
+from unittest import mock
+
+import sawtooth_signing as signing
+
+from sawtooth_poet.poet_consensus import poet_block_publisher
+
+from sawtooth_poet_common.protobuf.validator_registry_pb2 \
+    import ValidatorInfo
+from sawtooth_poet_common.protobuf.validator_registry_pb2 \
+    import SignUpInfo
+
+
+@mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.BlockWrapper')
+@mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.PoetConfigView')
+class TestPoetBlockPublisher(TestCase):
+
+    def setUp(self):
+        # pylint: disable=invalid-name,global-statement
+        global poet_block_publisher
+        # PoetBLockPublisher uses class variables to hold state
+        # so the module needs to be reloaded after each test to clear state
+        poet_block_publisher = reload(poet_block_publisher)
+
+        self._temp_dir = tempfile.mkdtemp()
+
+        # # PoetBLockPublisher uses class variables to hold state
+        # # so the module needs to be reloaded after each test to clear state
+        # # pylint: disable=invalid-name,global-statement
+        # global ploet_block_publisher
+        # ploet_block_publisher = reload(poet_block_publisher)
+
+    def tearDown(self):
+        shutil.rmtree(self._temp_dir)
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'SignupInfo')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'PoetKeyStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.utils')
+    def test_no_public_key(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_key_state_store,
+            mock_signup_info,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ Test verifies that PoET Block Publisher fails if
+        a validator's poet_public_key is not correct
```

Comment:
This test actually tests that the block publisher fails if it doesn't have any signup information in the validator registry.